optimum/onnxruntime/configuration.py

```diff
@@ -604,73 +608,248 @@ def tensorrt(
+
+    def create_fusion_options(self, model_type: str) -> FusionOptions:
+        class Box:
+            pass
+
+        args = Box()
+        args.model_type = model_type
+        attribute_map = {
+            "disable_gelu_fusion": "disable_gelu",
+            "disable_layer_norm_fusion": "disable_layer_norm",
+            "disable_attention_fusion": "disable_attention",
+            "disable_skip_layer_norm_fusion": "disable_skip_layer_norm",
+            "disable_bias_skip_layer_norm_fusion": "disable_bias_skip_layer_norm",
+            "disable_bias_gelu_fusion": "disable_bias_gelu",
+            "disable_embed_layer_norm_fusion": "disable_embed_layer_norm",
+        }
+        for attr_name, fusion_attr_name in attribute_map.items():
+            setattr(args, fusion_attr_name, getattr(self, attr_name))
+
+        for attr, value in self.__dict__.items():
+            if hasattr(args, attr):
+                continue
+            setattr(args, attr, value)
+
+        return FusionOptions.parse(args)
+
+
+class AutoOptimizationConfig:
+    """
+    Factory to create common `OptimizationConfig`.
+    """
+
+    _LEVELS = {
+        "O1": {
+            "optimization_level": 1,
+            "enable_transformers_specific_optimizations": False,
+        },
+        "O2": {
+            "optimization_level": 2,
+            "enable_transformers_specific_optimizations": True,
+            "enable_gelu_approximation": True,
+        },
+        "O3": {
+            "optimization_level": 2,
+            "enable_transformers_specific_optimizations": True,
+            "enable_gelu_approximation": True,
+            "fp16": True,
+        },
+    }
+
+    @classmethod
+    def with_optimization_level(cls, optimization_level: str, for_gpu: bool = False, **kwargs) -> OptimizationConfig:
+        """
+        Creates an [`~OptimizationConfig`] with pre-defined arguments according to an optimization level.
+
+        Args:
+            optimization_level (`str`):
+                The optimization level, the following values are allowed:
+                - O1: Basic general optimizations
+                - O2: Basic and extended general optimizations, transformers-specific fusions, and Gelu approximation.
+                - O3: Same as O2, with mixed precision.
+            for_gpu (`bool`, *optional*, defaults to `False`):
+                Whether the model to optimize will run on GPU, some optimizations depends on the hardware the model
+                will run on. Only needed for optimization_level > 1.
+            kwargs (`Dict[str, Any]`, *optional*):
+                Arguments to provide to the [`~OptimizationConfig`] constructor.
+
+        Returns:
+            `OptimizationConfig`: The `OptimizationConfig` corresponding to the requested optimization level.
+        """
+        if optimization_level not in cls._LEVELS:
+            raise ValueError(f"optimization_level must be in {', '.join(cls._LEVELS.keys())}")
```

Comment:
Nit: I would add at the end of the error message: "got {optimization_level}"
```suggestion
            raise ValueError(f"optimization_level must be in {', '.join(cls._LEVELS.keys())}, got {optimization_level}")
```